### DISCUSSION TITLE
Add support for the Zim wiki syntax

### DISCUSSION THREAD

**Discussion by radupotop** [GitJournal/GitJournal#682] [Feature Requests]
It would be amazing if GitJournal supported the Zim wiki syntax. 

[Zim][] is a simple wiki app available for Linux/Windows/Mac, it has an uncluttered & native interface. It allows you to write pages and group them in a hierarchical structure. The best feature is that it allows you to commit changes to a git repo straight from the UI.  
It works very similarly to GitJournal but on the desktop.

Bringing these two together would be an incredible achievement since you could now sync notes and wikis between mobile and desktop. I'm using it on Linux and have a bunch of git hooks to automatically push changes to Github.

The syntax it uses is the following: https://zim-wiki.org/manual/Help/Wiki_Syntax.html

## Notebook format

At the root folder Zim can create _notebooks_ for holding different topics.
Each notebook is a plain folder which has a `notebook.zim` file inside:

```
[Notebook]
version=0.4
name=MyNotes
interwiki=
home=Home
icon=
document_root=
shared=True
endofline=unix
disable_trash=False
```

Besides the `name=` variable I don't think there's much else you would typically need to modify.  

Each notebook folder can have files and subfolders.

## File format

Each file has a metadata header like so:

```
Content-Type: text/x-zim-wiki
Wiki-Format: zim 0.6
Creation-Date: 2021-03-02T21:48:36+00:00
```

Followed by a new line and the content itself, no other separator.

## Syntax not covered by the docs

I'm not sure whether the following syntax is in the docs.

### Inline code:
```
''inline code is denoted by two quotes''
```

### Code block / pre-formatted text:
```
'''
Code blocks are surrounded by groups of three quotes.
'''
```

### Link syntax

The most common link syntax is:

```
[[URL|LINK_LABEL]]
```

But it also supports internal links to other pages.

### Plugins

Zim supports lots of plugins. They usually extend the basic syntax with their own.
You have plugins for music scores, diagrams, math notation, etc. Some of these require different system libraries to be installed, etc.  
I don't think plugins should be a priority at all, just something to be aware of.

One plugin which I found particularly useful is the table plugin.  
The syntax looks like this:

```
| ID | Retailer | Branch        | Offer name                      | Offer external ID |
|:---|:---------|:--------------|:--------------------------------|:------------------|
| 1  | Argos    | Holloway      | Discounts on Gardening          | UUID4             |
| 2  | Tesco    | 100 East Road | 20% off on house cleaning items | UUID4             |
```

Which then renders into a nice native table.

Please let me know how I can help with more syntax examples, testing, or other things you need clarifying.

Cheers

[zim]: https://zim-wiki.org/index.html

---
> Please vote on this issue with a 👍 reaction. It's one of the primary mechanisms used to prioritize issues. Please also subscribe to the issue if you're interested, as that way I can discuss possible implementations and share mockups.

**Comment by vHanda** [comment]
The easiest way for me to add Zim Sytax support would be if there if someone could develop an automated test suite for the Zim language <-> AST. Ideally the AST could be represented in a language agnostic manner via ProtoBuf.
